controlplane/internal/controllers/rke2controlplane_controller.go

```diff
@@ -352,15 +353,14 @@ func (r *RKE2ControlPlaneReconciler) updateStatus(ctx context.Context, rcp *cont
 
 	availableCPMachines := readyMachines
 
-	validIPAddresses := []string{}
-
-	for _, machine := range availableCPMachines {
-		ipAddress, err := getIPAddress(*machine)
-		if err != nil {
-			break
-		}
+	registrationmethod, err := registration.NewRegistrationMethod(string(rcp.Spec.RegistrationMethod))
+	if err != nil {
+		return fmt.Errorf("getting node registration method: %w", err)
```

Comment:
```suggestion
		return fmt.Errorf("error getting node registration method: %w", err)
```

Reply:
If i'm wrapping error i generally don'y include "error" but instead give the context.